### ISSUE TITLE
Use SecTrustEvaluateWithError

### ISSUE THREAD

**Issue by stepancheg** [kornelski/rust-security-framework#114]
Current implementation of the crate uses `SecTrustEvaluate` which returns `The certificate was not trusted` which does not help much understanding what's wrong.

Documentation of [`SecTrustEvaluate`](https://developer.apple.com/documentation/security/1394363-sectrustevaluate?language=objc) suggests using `SecTrustEvaluateWithError` function, which (I suspect) provides more detailed error.

**Comment by stepancheg** [comment]
Yes, I tried to do some hacks, and the error became actually

```
“foobar.com” certificate name does not match input"
```

which is much better than

```
The certificate was not trusted
```

Trying to figure out how to patch security-framework crate.

**Comment by bascurtiz** [comment]
@stepancheg 
Can you please update crates.io too?
I confirm, your latest commit, v2.1.2 seems to fix it while v2.1.1 results in the `SecTrustEvaluateWithError` output, running macOS El Capitan (10.11) - Mojave (10.14) and above seems not affected.

Output log v2.1.2+ v2.1.1 attached.
[output-log-v2.1.2.txt](https://github.com/kornelski/rust-security-framework/files/6100582/output-log-v2.1.2.txt)
[output-log-v2.1.1.txt](https://github.com/kornelski/rust-security-framework/files/6100583/output-log-v2.1.1.txt)

**Comment by kornelski** [comment]
Published
